usaspending_api/disaster/v2/views/award/amount.py

```diff
@@ -44,22 +47,25 @@ def queryset(self):
             self.is_in_provided_def_codes,
         ]
 
-        if self.award_type_codes:
-            count_field = "award_id"
-        else:
-            count_field = self.unique_file_c_awards
+        group_by_annotations = {
+            "award_identifier": F("award_id") if self.award_type_codes else self.unique_file_c_awards
+        }
 
-        fields = {
-            "award_count": Count(count_field, distinct=True),
-            "obligation": Coalesce(Sum("total_obligation"), 0),
-            "outlay": Coalesce(Sum("total_outlay"), 0),
+        dollar_annotations = {
+            "inner_obligation": self.obligated_field_annotation,
+            "inner_outlay": self.outlay_field_annotation,
         }
 
-        if self.award_type_codes:
-            return self.when_non_zero_award_spending(
-                FinancialAccountsByAwards.objects.filter(*filters).values(count_field)
-            ).aggregate(**fields)
-        else:
-            return self.when_non_zero_award_spending(
-                FinancialAccountsByAwards.objects.filter(*filters).annotate(unique_c=count_field).values("unique_c")
-            ).aggregate(**fields)
+        all_annotations = {**group_by_annotations, **dollar_annotations}
+
+        return (
+            FinancialAccountsByAwards.objects.filter(*filters)
+            .annotate(**group_by_annotations)
+            .values(*group_by_annotations)
+            .annotate(**dollar_annotations)
+            .exclude(inner_obligation=0, inner_outlay=0)
```

Comment:
This looks like a great replacement to the clunky `when_non_zero_award_spending` later down the line.

Reply:
Have to be careful, though.  When I tested `exclude` several months, it expressed the same bug on annotations, but it's not here so I left it in.